src/m365/commands/login.spec.ts

```diff
@@ -299,32 +300,33 @@ describe(commands.LOGIN, () => {
   });
 
   it(`Don't start login flow when the CLI is signed in`, async () => {
-    sinon.stub(auth, 'ensureAccessToken').callsFake(() => Promise.resolve(''));
-
     await command.action(logger, { options: { ensure: true } });
-    assert.strictEqual(auth.service.connected, true);
+    await command.action(logger, { options: { ensure: true } });
+
+    assert(deactivateStub.callCount <= 1);
   });
 
   it(`Don't start login flow if the CLI is signed in as a user`, async () => {
-    sinon.stub(auth, 'ensureAccessToken').callsFake(() => Promise.resolve(''));
-
     await command.action(logger, { options: { ensure: true, authType: 'password', userName: 'john.doe@contoso.com', password: 'password' } });
-    assert.strictEqual(auth.service.connected, true);
+    await command.action(logger, { options: { ensure: true, authType: 'password', userName: 'john.doe@contoso.com', password: 'password' } });
+
+    assert(deactivateStub.callCount <= 1);
```

Comment:
If the user is already signed in, why should we call deactivate? This applies to all test cases where we've got `<= 1`

Reply:
I'm confused here 😄 We should check if deactivate (the previous logout) is not called.
if a user is logged in (when ´auth.connection.active´ is ´true´) deactivate won't be called, so callCount will be 0. If a user is not logged in (when ´auth.connection.active´ is ´false´) deactivate will be called once.

As we perform 2 command.action's in these tests the callcount will be 0 or 1, but can't be more then 1?

Reply:
These are two different test cases so should be implemented in two separate tests with precise assert.